ui/src/components/text-truncate/QasTextTruncate.vue

```diff
@@ -176,15 +181,42 @@ function useDialog ({ props, textContent }) {
     }
   })
 
+  const normalizedSearchModel = computed(() => {
+    return props.useObjectList ? searchModel.value.map(({ label }) => label) : searchModel.value
+  })
+
+  const hasSearchBox = computed(() => normalizedList.value.length > 12)
+
+  const dialogComponent = computed(() => {
```

Comment:
to achando que compensa criar um componente privado para o dialog como um todo, o que acha?

Reply:
já tá organizado com composable, se adicionarmos mais coisas, futuramente podemos criar